Source/Blazorise/Components/Carousel/Carousel.razor.cs

```diff
@@ -429,7 +408,17 @@ protected virtual async Task RunAnimations()
 
             await InvokeAsync( StateHasChanged );
 
-            ResetTransitionTimer();
+            await ResetTransitionTimer();
+
+            await Task.Delay( TimeSpan.FromMilliseconds( 2000 ) );
```

Comment:
Is this delay `await Task.Delay( TimeSpan.FromMilliseconds( 2000 ) );` for debugging only?

Reply:
Transition timer was running in a 2s  interval, so while not exactly. The same it's pretty close to what it was.... I guess I must further investigate why transition timer exists and is called on a 2s interval...

Reply:
I have tested and you're right. It doesn't pass on with initializing cycle. And I found another even worse problem. if you navigate away from the carousel page the timer is still called every time, even though it is disposed of.

Reply:
God damn it https://docs.microsoft.com/en-us/dotnet/api/system.threading.periodictimer.waitfornexttickasync?view=net-6.0#parameters

Reply:
So what about just returning to the previous background implementation?

Reply:
I think it is best to just not use PeriodicTimer and just leave it as it is now.

Reply:
Right, that also went through my mind.
Altough the implementation was starting to look a lot cleaner.